protoc-gen-gogo/generator/helper.go

```diff
@@ -181,10 +181,19 @@ func (g *Generator) GetFieldName(message *Descriptor, field *descriptor.FieldDes
 			return fieldname + "_"
 		}
 	}
+	if gogoproto.IsProtoSizer(file.FileDescriptorProto, message.DescriptorProto) {
```

Comment:
No my bad.  Maybe add ProtoSize as a reserved name, but yes we are trying to make Size not a reserved word.  So keep the check only for the Size method.